File: Part8/C++/FunctionConcepts/FunctionConcepts_11.cpp
// Recursive function call to compute factorial
#include <iostream>
#include <cstdio>
using namespace std;

int factorial ( int FactorialNumber_Par )
{
    if ( FactorialNumber_Par <= 1 )
    {
        return 1;
    }
    
    return FactorialNumber_Par * factorial ( FactorialNumber_Par - 1 );
}

int main ()
{
    int factorialResult = factorial ( 10 );
    printf ( "Factorial of 10 is: %d\n", factorialResult );
    return 0;
}

File: Part8/Rust/FunctionConcepts/FunctionConcepts_11.rs
// Recursive function call to compute factorial
fn factorial ( factorial_number_par: i32 ) -> i32
{
    if factorial_number_par <= 1
    {
        return 1;
    }
    
    return factorial_number_par * factorial ( factorial_number_par - 1 );
}

fn main ()
{
    let factorial_result = factorial ( 10 );
    println! ( "Factorial of 10 is: {}", factorial_result );
}
